images/common/collectstatic.py

```diff
@@ -0,0 +1,47 @@
+"""Call `collectstatic` only when dependencies have changed.
+
+This helps speed up startup on cloud platforms. To disable this behavior,
+set the `COLLECTSTATIC_WHEN_DEPS_CHANGE` environment variable to `False`.
+"""
+
+import hashlib
+import os
+import subprocess
+import sys
+
+import django
+import redis
+from django.conf import settings
+
+os.environ.setdefault("DJANGO_SETTINGS_MODULE", "openwisp.settings")
+django.setup()
+
+
+def get_pip_freeze_hash():
+    try:
+        pip_freeze_output = subprocess.check_output(["pip", "freeze"]).decode()
+        return hashlib.sha256(pip_freeze_output.encode()).hexdigest()
+    except subprocess.CalledProcessError as e:
+        print(f"Error running 'pip freeze': {e}", file=sys.stderr)
+        sys.exit(1)
+
+
+def main():
+    if os.environ.get("COLLECTSTATIC_WHEN_DEPS_CHANGE", "true").lower() == "false":
+        print("COLLECTSTATIC_WHEN_DEPS_CHANGE is false; skipping collectstatic.")
```

Comment:
This is not what I meant.
If the env var is False we shall always execute `collectstatic`. **Not executing it is not an option**.